Game.js

```diff
@@ -0,0 +1,279 @@
+class Utils {
+  // Preset character colors.
+  static colors = ['red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet'];
+
+  // Gives a random number between two numbers.
+  static randomizeRange(floor, ceil) {
+    return Math.floor(Math.random() * ceil) + floor;
+  }
+
+  // This method makes it impossible for an enemy to spawn on the player and also makes sure they spawn in the area.
+  static calculateSpawn(player, boundaries) {
+    const playerPos = player.position;
+    const pos = { x: null, y: null };
+    // Space available.
+    const leftSpace = playerPos.x;
+    const rightSpace = boundaries.maxX - playerPos.x;
+    const upSpace = playerPos.y;
+    const downSpace = boundaries.maxY - playerPos.y;
+    // Side with more space.
+    const dominantX = leftSpace >= rightSpace ? 'left' : 'right';
+    const dominantY = upSpace >= downSpace ? 'up' : 'down';
+    const dominantXDist = Math.max(leftSpace, rightSpace);
+    const dominantYDist = Math.max(upSpace, downSpace);
+    // How far from player is the enemy going to spawn, minimum is the width/height of player.
+    const xSpawnDist = Utils.randomizeRange(player.element.offsetWidth, dominantXDist - player.element.offsetWidth);
+    const ySpawnDist = Utils.randomizeRange(player.element.offsetHeight, dominantYDist - player.element.offsetHeight);
+    // Calculating the spawn coords.
+    if (dominantX === 'left') pos.x = playerPos.x - xSpawnDist;
+    if (dominantX === 'right') pos.x = playerPos.x + xSpawnDist;
+    if (dominantY === 'up') pos.y = playerPos.y - ySpawnDist;
+    if (dominantY === 'down') pos.y = playerPos.y + ySpawnDist;
+    return pos;
+  }
```

Comment:
Awesome user experience! I noticed that the spawn locations seemed well calculated while I was playing!